script.js

```diff
@@ -1,4 +1,160 @@
+    currentMode = "normal mode";
+  } else if (currentMode == "start" && input == "reversed") {
+    myOutputValue = `You have chosen REVERSED mode. <br><br> Press enter to start rolling the dice 🎲`;
+    currentMode = "normal mode";
+    reversedMode = true;
+    // normal mode will roll the dice
+  } else if (currentMode == "normal mode") {
+    diceRoll1Player1 = diceRollGenerator();
+    diceRoll2Player1 = diceRollGenerator();
+    myOutputValue = player1RollDice();
+    currentMode = "player 1 rolled";
+    // input the order
+  } else if (currentMode == "player 1 rolled" && input == 1) {
+    var player1Order = diceOrder1Player1();
+    currentMode = "player 2 start";
+    return player1Order;
+  } else if (currentMode == "player 1 rolled" && input == 2) {
+    var player1Order = diceOrder2Player1();
+    currentMode = "player 2 start";
+    return player1Order;
+
+    // player 2 roll the dice
+  } else if (currentMode == "player 2 start") {
+    diceRoll1Player2 = diceRollGenerator();
+    diceRoll2Player2 = diceRollGenerator();
+    myOutputValue = player2RollDice();
+    currentMode = "player 2 rolled";
+  } else if (currentMode == "player 2 rolled" && input == 1) {
+    var player2Order = diceOrder1Player2();
+    currentMode = "results";
+    return player2Order;
+  } else if (currentMode == "player 2 rolled" && input == 2) {
+    var player2Order = diceOrder2Player2();
+    currentMode = "results";
+    return player2Order;
+
+    // shows result
+  } else if (currentMode == "results" && player1FinalNumb == player2FinalNumb) {
+    myOutputValue = `Its a draw. <br><br> Player 1: ${player1FinalNumb} <br><br> Player 2: ${player2FinalNumb}`;
+    currentMode = "leaderboard";
+  } else if (
+    currentMode == "results" &&
+    player1FinalNumb > player2FinalNumb &&
+    reversedMode == false
+  ) {
+    player1Score++;
+    currentMode = "leaderboard";
+    myOutputValue = `Player 1 wins!<br><br> Player 1: ${player1FinalNumb} <br><br> Player 2: ${player2FinalNumb}`;
+  } else if (
+    currentMode == "results" &&
+    player1FinalNumb < player2FinalNumb &&
+    reversedMode == false
+  ) {
+    player2Score++;
+    myOutputValue = `Player 2 wins!<br><br> Player 1: ${player1FinalNumb} <br><br> Player 2: ${player2FinalNumb}`;
+    currentMode = "leaderboard";
+  } else if (
+    currentMode == "results" &&
+    reversedMode == true &&
+    player1FinalNumb < player2FinalNumb
+  ) {
+    player1Score++;
+    currentMode = "leaderboard";
+    myOutputValue = `Player 1 wins!<br><br> Player 1: ${player1FinalNumb} <br><br> Player 2: ${player2FinalNumb}`;
+  } else if (
+    currentMode == "results" &&
+    reversedMode == true &&
+    player1FinalNumb > player2FinalNumb
+  ) {
+    player2Score++;
+    myOutputValue = `Player 2 wins!<br><br> Player 1: ${player1FinalNumb} <br><br> Player 2: ${player2FinalNumb}`;
+    currentMode = "leaderboard";
+  }
```

Comment:
Same as the above comments with regards to combining the various code for the same game mode into a helper function!